File: native/Travel Companion/Model/Firestore/Plannable.swift
//
//  Plannable.swift
//  Travel Companion
//
//  Created by Stefan Jaindl on 12.09.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//

import Foundation

protocol Plannable: Codable {
    func description() -> String
    func details() -> NSMutableAttributedString
    func imageUrl() -> String?
    func getId() -> String
    func getLink() -> String?
    func getLinkText() -> NSMutableAttributedString?
    func getNotes() -> String
    func setNotes(notes: String)
    func encode() -> [String: Any]
}

extension Plannable {
    func getLink() -> String? {
        return nil
    }
    
    func getLinkText() -> NSMutableAttributedString? {
        return nil
    }
}

File: shared/src/commonMain/kotlin/com.sjaindl.travelcompanion/api/Plannable.kt
package com.sjaindl.travelcompanion.api

interface Plannable {
    fun description(): String

    // fun details(): NSMutableAttributedString
    // fun imageUrl(): String?
    fun getId(): String
    // fun getLink(): String?

    // fun getLinkText(): NSMutableAttributedString?
    // fun getNotes(): String
    // fun setNotes(notes: String)
    // fun encode(): Map<String, Any>
}

// TODO:
/*
internal fun Plannable.getLink(): String? =
    null

internal fun Plannable.getLinkText(): NSMutableAttributedString? =
    null
 */
